src/api/routers/participants/participantsCreation.ts

```diff
@@ -142,11 +172,59 @@ export async function createParticipant(req: ParticipantRequest, res: Response)
     );
 
     // assign proper api access
-    assignClientRoleToUser(kcAdminClient, newUser!.email, 'api-participant-member');
+    assignClientRoleToUser(kcAdminClient, user.email, 'api-participant-member');
 
     // send email
     await sendInviteEmail(kcAdminClient, newKcUser);
   });
 
   return res.sendStatus(200);
 }
+
+export const createParticipantFromRequest = async (req: ParticipantRequest, res: Response) => {
```

Comment:
Extracted this method from `participantsRouter.ts`. Similar changes to `createParticipant` but decided against factoring out similar code because of slight differences in the data they would pass in and return.